v-next/hardhat-ignition/src/internal/tasks/migrate.ts

```diff
@@ -0,0 +1,96 @@
+import type {
+  Artifact,
+  BuildInfo,
+  CompilerOutput,
+} from "@nomicfoundation/ignition-core";
+import type { HardhatRuntimeEnvironment } from "hardhat/types/hre";
+import type { NewTaskActionFunction } from "hardhat/types/tasks";
+
+import path from "node:path";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+import {
+  exists,
+  readdir,
+  readJsonFile,
+  remove,
+  writeJsonFile,
+} from "@nomicfoundation/hardhat-utils/fs";
+
+interface MigrateArguments {
+  deploymentId: string;
+}
+
+const taskMigrate: NewTaskActionFunction<MigrateArguments> = async (
+  { deploymentId },
+  hre: HardhatRuntimeEnvironment,
+) => {
+  const deploymentDir = path.join(
+    hre.config.paths.ignition,
+    "deployments",
+    deploymentId,
+  );
+
+  if (!(await exists(deploymentDir))) {
+    throw new HardhatError(
+      HardhatError.ERRORS.IGNITION.TRACK_TRANSACTIONS.DEPLOYMENT_DIR_NOT_FOUND,
+      {
+        deploymentDir,
+      },
+    );
+  }
+
+  const artifactsDir = `${deploymentDir}/artifacts`;
+
+  for (const filename of await readdir(artifactsDir)) {
+    if (filename.endsWith(".dbg.json")) {
+      continue;
+    }
+
+    const artifactPath = path.join(artifactsDir, filename);
+    const artifact: Artifact = await readJsonFile(artifactPath);
+
+    const debugFilePath = path.join(
+      artifactsDir,
+      `${filename.split(".json")[0]}.dbg.json`,
+    );
+    const debugFile: { buildInfo: string } = await readJsonFile(debugFilePath);
```

Comment:
I ran into a problem. In the complete example, libraries don't seem to generate `dbg.json` files.